### PR TITLE
Remove broken package URL validation

### PR DESCRIPTION
**Pull request by danibachar**
Resolves https://github.com/tuist/tuist/issues/YYY

### Short description 📝

The new URL API should be used by swift version and not OS.

### How to test the changes locally 🧐

Using the if #available(macOS 14.0, *) is not exhaustive in many cases like running macos 13 (ventura) with xcode 15. 
We should check the available Swift version which expose this new API as part of the Foundation framework.

I have run into this issue when building advanced versions (3.28.0 and above) of Tuist for Homebrew

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `make workspace/lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistSupport/Extensions/URL+Extras.swift [hunk 1]
```diff
@@ -2,15 +2,17 @@ import Foundation
 
 extension URL {
     public static func isValid(_ url: String) -> Bool {
-        var valid = false
         /**
          Xcode 15 introduced breaking changes in the URL constructor that we need to account for.
          */
         if #available(macOS 14.0, *) {
```
Comment (danyf90) [inline Sources/TuistSupport/Extensions/URL+Extras.swift:8]: We don't need this if, do we?

File: Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        XCTAssertTrue(URL.isValid("wss://google.com:443"))
+        XCTAssertTrue(URL.isValid("google.com"))
+        XCTAssertTrue(URL.isValid("google"))
+        XCTAssertTrue(URL.isValid("1"))
+    }
+    
+    func test_non_valid_url_empty() {
+        XCTAssertFalse(URL.isValid(""))
+    }
+}
```
Comment (danyf90) [inline Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift:24]: This is not supposed to be merged, right?

Comment (danibachar) [reply Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift:24]: I didn't see any tests for this API and thought of adding those, I can definitely remove them / migrate to another validation approach

**Comment by danibachar** [comment]
Going to close this PR, after testing it seems to not really solve the issue.
Xcode 15.0 has some kind of limitations for the new API usage

File: Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift (hunk 2 above)
Comment (danyf90) [reply Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift:24]: I would rather just remove this validation entirely as it looks quite useless :D

Comment (danyf90) [reply Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift:24]: Or use a regex as you said, but probably not worth the effort

Comment (danibachar) [reply Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift:24]: I tend to agree .
If its ok with you, I'm going with your first suggestion of removing it altogether. This validation is only used as part of the linter under `Sources/TuistGenerator/Linter/PackageLinter.swift`.

My line of thought here is that SPM supports different types of URLs that i'm not even sure Ill be able to cover (https/http/git@ for ssh etc)

I think that delegating the verification to SPM is the best approach

**Review by fortmarek — APPROVED** [review]

**Review by danyf90 — APPROVED** [review]

**Pull request merged**